py_gtfs_rt_ingestion/py_gtfs_rt_ingestion/convert.py

```diff
@@ -11,42 +10,34 @@
 def gz_to_pyarrow(filename: str, config: Configuration):
     """
     Accepts filename as string. Converts gzipped json -> pyarrow table.
+
+    Will handle Local of S3 filenames
     """
     logging.info("Converting %s to Parquet Table" % filename)
     try:
-        with gzip.open(filename, 'rb') as f:
-            json_data = json.load(f)
-            pa_table = _json_to_pyarrow(json_data=json_data, config=config)
-
-        return pa_table
+        if str(filename).startswith('s3'):
+            active_fs = fs.S3FileSystem()
```

Comment:
On my system, locally, the `S3FileSystem()` call takes quite a bit of time. Not sure if it would be thread safe to pass a shared version of this object around to the different threads, but could be an optimization step.

Reply:
i tried to pass a shared version earlier and ran into threading issues. that said, i'm sure they can be sorted out.